scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
+		body = src.(map[string]interface{})
+	}
+
 	// Search
 	path += "/_search"
 
 	// Parameters
 	params := make(url.Values)
-	if !s.searchSource.hasSort() {
+	if _, ok := body["sort"]; !ok {
```

Comment:
And things like these are the core of my problem with your approach: You pick one thing (e.g. `sort`) and fix that. But you open up a can of worms. In the near future I'll get an issue that says something like: "I have set the body of the request via `Source` and now I only want to change this or that with the DSL. Why the heck can't I do this?" That's why I chose to make it one-way. Use source? You're on your own.

Nevermind. Back to the review...

We should simply remove the `search_type=scan` here. It's deprecated and users should simply [specify `"sort":["_doc"]`](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/breaking_21_search_changes.html#_literal_search_type_scan_literal_deprecated) when using `Source` (we could still do that automatically when users chose the DSL via `SearchSource`, but maybe ES already does that automatically--will have to check my sources). Also, you'd no longer be required to do the type conversion mentioned earlier.